pkg/utils/resources/resources.go

```diff
@@ -97,13 +97,16 @@ func Subtract(lhs, rhs v1.ResourceList) v1.ResourceList {
 func Ceiling(pod *v1.Pod) v1.ResourceRequirements {
 	var resources v1.ResourceRequirements
 	for _, container := range pod.Spec.Containers {
-		resources.Requests = Merge(resources.Requests, MergeResourceLimitsIntoRequests(container))
-		resources.Limits = Merge(resources.Limits, container.Resources.Limits)
+		resources.Requests = MergeInto(resources.Requests, MergeResourceLimitsIntoRequests(container))
+		resources.Limits = MergeInto(resources.Limits, container.Resources.Limits)
 	}
 	for _, container := range pod.Spec.InitContainers {
 		resources.Requests = MaxResources(resources.Requests, MergeResourceLimitsIntoRequests(container))
 		resources.Limits = MaxResources(resources.Limits, container.Resources.Limits)
 	}
+	if pod.Spec.Overhead != nil {
+		resources.Requests = MergeInto(resources.Requests, pod.Spec.Overhead)
```

Comment:
Do we need the `nil` check or can we always merge the overhead into the requests even if it's `nil`?

Reply:
Yeah, we could skip it and it would still work. I like the explicit check, you don't ever need to worry about MergeInto changing in some way that makes it no longer valid to pass a nil, but it could go either way.